packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -706,10 +704,6 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
       // by the deps optimizer
       if (config.server.preTransformRequests && staticImportedUrls.size) {
         staticImportedUrls.forEach(({ url, id }) => {
-          url = unwrapId(removeImportQuery(url)).replace(
-            NULL_BYTE_PLACEHOLDER,
-            '\0'
-          )
```

Comment:
I think this is needed. `transformRequest` expects an unwrapped+`\0` URL (we need a name for the two URL spaces, rollup URL and browser URL?). This may not be related to what #6390 tried to fix, but I think we should always call `transformRequest` with a rollup URL. Probably not in this PR, but for 3.1 it may be worth allowing `transformRequest` to accept an URL in any space and always call unwrap+removeImportQuery+use`\0` (maybe we could have a `toRollupURL()` helper that does this? Still unsure about naming).